sycl/source/detail/program_impl.cpp

```diff
@@ -184,9 +184,6 @@ program_impl::program_impl(ContextImplPtr Context,
       OptionsVector.data(), nullptr);
   std::string Options(OptionsVector.begin(), OptionsVector.end());
   switch (BinaryType) {
-  case PI_PROGRAM_BINARY_TYPE_NONE:
```

Comment:
This is a good diagnostic to find errors during some code refactoring, now it is removed, so suggest adding the default case and add assert(false) in it.

Reply:
Good suggestion but basically this "Logically dead code" is because there is code to check if BinaryType is PI_PROGRAM_BINARY_TYPE_NONE and throw an exception in line 173. Coverity reported case PI_PROGRAM_BINARY_TYPE_NONE is redundancy and logically dead code.

173  if (BinaryType == PI_PROGRAM_BINARY_TYPE_NONE) {
174    throw invalid_object_error(

Reply:
Still it is helpful for us during the refactoring if the exception is removed

Reply:
Added a default case as advised

Reply:
Maybe do
```
switch () {
case 1:
 ...
 return;
case 2:
case 3:
  ...
  return;
}
assert(false) or __builtin_unreachable();
```

instead?

Reply:
It seems the BinaryType value is invalid and switch-default addition caught it. Under investigating.

# command stderr:
online_compiler_L0.cpp.tmp.out: /__w/llvm/llvm/src/sycl/source/detail/program_impl.cpp:200: sycl::_V1::detail::program_impl::program_impl(sycl::_V1::detail::ContextImplPtr, pi_native_handle, sycl::_V1::detail::pi::PiProgram): Assertion `false && "BinaryType is invalid."' failed.